scripts/cleanup_failed_registrations.py

```diff
@@ -31,6 +31,8 @@ def remove_failed_registrations(dry_run=True):
     if not dry_run:
         for f in failed:
             logging.info('Cleaning {}'.format(f))
+            if not f.registered_from:
```

Comment:
This should never happen, correct? 

If it does, it's unexpected, and we should log it